src/code42cli/cmds/devices.py

```diff
@@ -0,0 +1,399 @@
+)
+
+
+@devices.command()
+@device_id_option
+@change_device_name_option
+@purge_date_option
+@sdk_options()
+def deactivate(state, device_id, change_device_name, purge_date):
+    """Deactivate a device within Code42"""
+    _deactivate_device(state.sdk, device_id, change_device_name, purge_date)
+
+
+def _deactivate_device(sdk, device_id, change_device_name, purge_date):
+    try:
+        sdk.devices.deactivate(device_id)
+    except exceptions.Py42BadRequestError:
+        raise Code42CLIError("The device {} is in legal hold.".format(device_id))
+    except exceptions.Py42NotFoundError:
+        raise Code42CLIError("The device {} was not found.".format(device_id))
+    except exceptions.Py42ForbiddenError:
+        raise Code42CLIError("Unable to deactivate {}.".format(device_id))
+    if purge_date or change_device_name:
+        device = sdk.devices.get_by_id(device_id)
+        guid = device.data["guid"]
+        name = device.data["name"]
+    if purge_date:
+        _update_cold_storage_purge_date(sdk, guid, purge_date)
+    if change_device_name and not name.startswith("deactivated_"):
+        _change_device_name(
+            sdk, guid, "deactivated_" + date.today().strftime("%Y-%m-%d") + "_" + name
+        )
+
+
+def _update_cold_storage_purge_date(sdk, guid, purge_date):
+    archives_response = sdk.archive.get_all_by_device_guid(guid)
+    archive_guid_list = [
+        archive["archiveGuid"]
+        for page in archives_response
+        for archive in page["archives"]
+    ]
+    for archive_guid in archive_guid_list:
+        sdk.archive.update_cold_storage_purge_date(archive_guid, purge_date)
+
+
+def _change_device_name(sdk, guid, name):
+    device_settings = sdk.devices.get_settings(guid)
+    device_settings.name = name
+    sdk.devices.update_settings(device_settings)
+
+
+_DEVICE_INFO_KEYS_MAP = OrderedDict()
+_DEVICE_INFO_KEYS_MAP["computerId"] = "Device ID"
+_DEVICE_INFO_KEYS_MAP["name"] = "Name"
+_DEVICE_INFO_KEYS_MAP["osHostname"] = "Hostname"
+_DEVICE_INFO_KEYS_MAP["guid"] = "GUID"
+_DEVICE_INFO_KEYS_MAP["status"] = "Status"
+_DEVICE_INFO_KEYS_MAP["archiveBytes"] = "Largest Archive Size in Bytes"
+_DEVICE_INFO_KEYS_MAP["lastConnected"] = "Last Connected Date"
+_DEVICE_INFO_KEYS_MAP["lastBackup"] = "Last Backup Date"
+_DEVICE_INFO_KEYS_MAP["lastCompletedBackup"] = "Last Completed Backup Date"
+_DEVICE_INFO_KEYS_MAP["productVersion"] = "Code42 Version"
+_DEVICE_INFO_KEYS_MAP["osName"] = "Operating System"
+_DEVICE_INFO_KEYS_MAP["osVersion"] = "Operating System Version"
+
+
+@devices.command()
+@device_id_option
+@format_option
+@sdk_options()
+def show(state, device_id, format=None):
+    """Print device info."""
+    formatter = OutputFormatter(format, _DEVICE_INFO_KEYS_MAP)
+    device_info = _get_device_info(state.sdk, device_id)
+    if device_info:
```

Comment:
What is the situation when device_info is falsy?

Reply:
I was thinking it might be if we didn't get device info, but we should still have the response even if it's empty. Removed this conditional.